### PR TITLE
Accountability table cleanup

### PR DESCRIPTION
**Pull request by crast**
* Create a new table for accountability->person with:
  - An actual ID primary key to be able to manage duplicates
  - A center_id column to capture the associated center ID
  - Some fixes on older now-invalid SQL defaults.
* Create a migration script to move the data over to new format
* Update using simpler ORM commands instead of raw SQL

### CODE REVIEW

File: src/app/AccountabilityMapping.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            // For any remaining people holding this accountability, curtail them (shorten)
+            foreach ($accExisting->flatten(1) as $item) {
+                if ($item->starts_at->lt($startsAt)) {
+                    $item->ends_at = $startsAt;
+                    $item->save();
+                    $report['shortened']++;
+                } else if ($item->ends_at->lte($endsAt)) {
+                    // Fully enclosed within this time range, can delete.
+                    $report['deleted']++;
+                    $item->delete();
```
Comment (pdarg) [inline src/app/AccountabilityMapping.php]: think it's worth soft-deleting these so we don't loose that they were accountable? This is a pretty cornery corner -case

Comment (crast) [reply src/app/AccountabilityMapping.php]: If I enable soft deletes, then the `UNIQUE` constraint can't be done.. mysql can't do a UNIQUE constraint that ignores soft-deleted things.

Comment (pdarg) [reply src/app/AccountabilityMapping.php]: ah, ok. not really worried about it since this should never really come up except in resubmit cases (I think)

**Comment by crast** [comment]
[Accountabilities table does random-seeming things](https://trello.com/c/mEGomfIC/300-accountabilities-table-does-random-seeming-things)

**Pull request merged**